src/App.vue

```diff
@@ -573,6 +626,20 @@ function logError(message: string) {
             <div id="renderOutput" v-show="currentDisplayMode == 'imageMain'">
                 <RenderCanvas :device="device" @log-error="logError" @log-output="(log) => { printedText = log }"
                     ref="renderCanvas"></RenderCanvas>
+                <div class="control-bar">
+                    <div class="controls-left">
+                        <button @click="resetFrame" title="Reset frame to 0">&#x23EE;</button>
+                        <button @click="prevFrame" title="Step backward">&#x23F4;</button>
+                        <button @click="nextFrame" title="Step forward">&#x23F5;</button>
+                        <button @click="togglePause" :title="isPaused ? 'Resume' : 'Pause'">⏯</button>
+                        <span class="frame-counter">Frame: {{ frameID }}</span>
+                        <span class="fps-counter">FPS: {{ fps }}</span>
+                        <span class="resolution">Resolution: {{ canvasWidth }}x{{ canvasHeight }}</span>
+                    </div>
+                    <div class="controls-right">
+                        <button @click="toggleFullscreen" title="Toggle full screen">&#x26F6;</button>
+                    </div>
+                </div>
```

Comment:
Why not move this to be inside of `RenderCanvas.vue`? That way you don't need to check `renderCanvas.value` in all the functions and don't need to mirror the variables.

Reply:
Should be migrated over now to RenderCanvas.vue.